### PR TITLE
Base code for Android support for Home Edge

### PR DESCRIPTION
**Pull request by suresh-lc**
Adding base code for Android support for Home Edge.

Features supported:
1. Device discovery
2. Service discovery
3. Service offload

Signed-off-by: suresh-lc <suresh.lc@samsung.com>

### CODE REVIEW

**Review by tiokim — APPROVED** [review]
Wonderful! Thank you for the great contribution!!

**Comment by tiokim** [comment]
@suresh-lc What about excluding `.idea` folder if it's inessential?

File: README.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# Home-Edge-Android
+
+As an continous process to support Home Edge (Linux) to multiple platforms, this project caters to support Home Edge for Android. There are various smart home devices and Android has been making its presence in many of these devices. The most essential part of the ecosystem has been Smart phone and specifically Android based smart phones. There are large number of applications developed to support user needs across various domains. Some of the service applications would need compute intensive or containerized applications which are not available on the Android phones. This project helps the service applications on the android phones to leverage on the service offered by devices in the same network.
```
Comment (MoonkiHong) [inline README.md]:
```suggestion
As a continous process to support Home Edge (Linux) to multiple platforms, this project caters to support Home Edge for Android. There are various smart home devices and Android has been making its presence in many of these devices. The most essential part of the ecosystem has been Smart phone and specifically Android based smart phones. There are large number of applications developed to support user needs across various domains. Some of the service applications would need compute intensive or containerized applications which are not available on the Android phones. This project helps the service applications on the android phones to leverage on the service offered by devices in the same network.
```

File: app/src/androidTest/java/org/lfedge/homeedge/ExampleInstrumentedTest.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
+package org.lfedge.homeedge;
```
Comment (MoonkiHong) [inline app/src/androidTest/java/org/lfedge/homeedge/ExampleInstrumentedTest.java:18]: Copyright statement is required.

File: app/src/main/java/org/lfedge/homeedge/HomeEdgeService.java [hunk 3]
```diff
@@ -0,0 +1,145 @@
+/****************************************************************************
+ Copyright 2022-2023 Samsung Electronics All Rights Reserved.
```
Comment (MoonkiHong) [inline app/src/main/java/org/lfedge/homeedge/HomeEdgeService.java]:
```suggestion
 Copyright 2022 Samsung Electronics All Rights Reserved.
```

File: app/src/main/java/org/lfedge/homeedge/MainActivity.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
+/****************************************************************************
+ Copyright 2022-2023 Samsung Electronics All Rights Reserved.
```
Comment (MoonkiHong) [inline app/src/main/java/org/lfedge/homeedge/MainActivity.java]:
```suggestion
 Copyright 2022 Samsung Electronics All Rights Reserved.
```

File: app/src/main/java/org/lfedge/homeedge/NsdHelper.java [hunk 5]
```diff
@@ -0,0 +1,253 @@
+/****************************************************************************
+ Copyright 2022-2023 Samsung Electronics All Rights Reserved.
```
Comment (MoonkiHong) [inline app/src/main/java/org/lfedge/homeedge/NsdHelper.java]:
```suggestion
 Copyright 2022 Samsung Electronics All Rights Reserved.
```

File: app/src/main/java/org/lfedge/homeedge/client/APIImpl.java [hunk 6]
```diff
@@ -0,0 +1,132 @@
+/****************************************************************************
+ Copyright 2022-2023 Samsung Electronics All Rights Reserved.
```
Comment (MoonkiHong) [inline app/src/main/java/org/lfedge/homeedge/client/APIImpl.java]:
```suggestion
 Copyright 2022 Samsung Electronics All Rights Reserved.
```

File: app/src/main/java/org/lfedge/homeedge/client/APIListener.java [hunk 7]
```diff
@@ -0,0 +1,22 @@
+/****************************************************************************
+ Copyright 2022-2023 Samsung Electronics All Rights Reserved.
```
Comment (MoonkiHong) [inline app/src/main/java/org/lfedge/homeedge/client/APIListener.java]:
```suggestion
 Copyright 2022 Samsung Electronics All Rights Reserved.
```

File: app/src/main/java/org/lfedge/homeedge/client/Api.java [hunk 8]
```diff
@@ -0,0 +1,37 @@
+/****************************************************************************
+ Copyright 2022-2023 Samsung Electronics All Rights Reserved.
```
Comment (MoonkiHong) [inline app/src/main/java/org/lfedge/homeedge/client/Api.java]:
```suggestion
 Copyright 2022 Samsung Electronics All Rights Reserved.
```

File: app/src/main/java/org/lfedge/homeedge/client/Client.java [hunk 9]
```diff
@@ -0,0 +1,46 @@
+/****************************************************************************
+ Copyright 2022-2023 Samsung Electronics All Rights Reserved.
```
Comment (MoonkiHong) [inline app/src/main/java/org/lfedge/homeedge/client/Client.java]:
```suggestion
 Copyright 2022 Samsung Electronics All Rights Reserved.
```

File: app/src/main/java/org/lfedge/homeedge/client/Response.java [hunk 10]
```diff
@@ -0,0 +1,68 @@
+/****************************************************************************
+ Copyright 2022-2023 Samsung Electronics All Rights Reserved.
```
Comment (MoonkiHong) [inline app/src/main/java/org/lfedge/homeedge/client/Response.java]:
```suggestion
 Copyright 2022 Samsung Electronics All Rights Reserved.
```

File: app/src/main/java/org/lfedge/homeedge/common/Utils.java [hunk 11]
```diff
@@ -0,0 +1,137 @@
+/****************************************************************************
+ Copyright 2022-2023 Samsung Electronics All Rights Reserved.
```
Comment (MoonkiHong) [inline app/src/main/java/org/lfedge/homeedge/common/Utils.java]:
```suggestion
 Copyright 2022 Samsung Electronics All Rights Reserved.
```

File: app/src/main/java/org/lfedge/homeedge/database/DeviceDao.java [hunk 12]
```diff
@@ -0,0 +1,55 @@
+/****************************************************************************
+ Copyright 2022-2023 Samsung Electronics All Rights Reserved.
```
Comment (MoonkiHong) [inline app/src/main/java/org/lfedge/homeedge/database/DeviceDao.java]:
```suggestion
 Copyright 2022 Samsung Electronics All Rights Reserved.
```

File: app/src/main/java/org/lfedge/homeedge/database/DeviceDatabase.java [hunk 13]
```diff
@@ -0,0 +1,56 @@
+/****************************************************************************
+ Copyright 2022-2023 Samsung Electronics All Rights Reserved.
```
Comment (MoonkiHong) [inline app/src/main/java/org/lfedge/homeedge/database/DeviceDatabase.java]:
```suggestion
 Copyright 2022 Samsung Electronics All Rights Reserved.
```

File: app/src/main/java/org/lfedge/homeedge/model/Device.java [hunk 14]
```diff
@@ -0,0 +1,102 @@
+/****************************************************************************
+ Copyright 2022-2023 Samsung Electronics All Rights Reserved.
```
Comment (MoonkiHong) [inline app/src/main/java/org/lfedge/homeedge/model/Device.java]:
```suggestion
 Copyright 2022 Samsung Electronics All Rights Reserved.
```

File: app/src/main/java/org/lfedge/homeedge/model/ListConverter.java [hunk 15]
```diff
@@ -0,0 +1,40 @@
+/****************************************************************************
+ Copyright 2022-2023 Samsung Electronics All Rights Reserved.
```
Comment (MoonkiHong) [inline app/src/main/java/org/lfedge/homeedge/model/ListConverter.java]:
```suggestion
 Copyright 2022 Samsung Electronics All Rights Reserved.
```

File: app/src/main/java/org/lfedge/homeedge/orchestrationapi/ExternalHandler.java [hunk 16]
```diff
@@ -0,0 +1,177 @@
+/****************************************************************************
+ Copyright 2022-2023 Samsung Electronics All Rights Reserved.
```
Comment (MoonkiHong) [inline app/src/main/java/org/lfedge/homeedge/orchestrationapi/ExternalHandler.java]:
```suggestion
 Copyright 2022 Samsung Electronics All Rights Reserved.
```

File: app/src/main/java/org/lfedge/homeedge/orchestrationapi/InternalHandler.java [hunk 17]
```diff
@@ -0,0 +1,73 @@
+/****************************************************************************
+ Copyright 2022-2023 Samsung Electronics All Rights Reserved.
```
Comment (MoonkiHong) [inline app/src/main/java/org/lfedge/homeedge/orchestrationapi/InternalHandler.java]:
```suggestion
 Copyright 2022 Samsung Electronics All Rights Reserved.
```

File: app/src/main/java/org/lfedge/homeedge/server/HTTPServer.java [hunk 18]
```diff
@@ -0,0 +1,65 @@
+/****************************************************************************
+ Copyright 2022-2023 Samsung Electronics All Rights Reserved.
```
Comment (MoonkiHong) [inline app/src/main/java/org/lfedge/homeedge/server/HTTPServer.java]:
```suggestion
 Copyright 2022 Samsung Electronics All Rights Reserved.
```

File: app/src/main/java/org/lfedge/homeedge/server/ServiceInfo.java [hunk 19]
```diff
@@ -0,0 +1,68 @@
+/****************************************************************************
+ Copyright 2022-2023 Samsung Electronics All Rights Reserved.
```
Comment (MoonkiHong) [inline app/src/main/java/org/lfedge/homeedge/server/ServiceInfo.java]:
```suggestion
 Copyright 2022 Samsung Electronics All Rights Reserved.
```

File: app/src/test/java/org/lfedge/homeedge/ExampleUnitTest.java [hunk 20]
```diff
@@ -0,0 +1,17 @@
+package org.lfedge.homeedge;
```
Comment (MoonkiHong) [inline app/src/test/java/org/lfedge/homeedge/ExampleUnitTest.java:18]: Copyright statement is required.

**Review by MoonkiHong — COMMENTED** [review]
Please take a look at my comments.

File: README.md [hunk 21]
```diff
@@ -0,0 +1,35 @@
+# Home-Edge-Android
+
+As a continous process to support Home Edge (Linux) to multiple platforms, this project caters to support Home Edge for Android. There are various smart home devices and Android has been making its presence in many of these devices. The most essential part of the ecosystem has been Smart phone and specifically Android based smart phones. There are large number of applications developed to support user needs across various domains. Some of the service applications would need compute intensive or containerized applications which are not available on the Android phones. This project helps the service applications on the android phones to leverage on the service offered by devices in the same network.
```
Comment (MoonkiHong) [inline README.md:3]:
```suggestion
As a continuous process to support Home Edge (Linux) to multiple platforms, this project caters to support Home Edge for Android. There are various smart home devices and Android has been making its presence in many of these devices. The most essential part of the ecosystem has been Smart phone and specifically Android based smart phones. There are large number of applications developed to support user needs across various domains. Some of the service applications would need compute intensive or containerized applications which are not available on the Android phones. This project helps the service applications on the android phones to leverage on the service offered by devices in the same network.
```

File: README.md [hunk 22]
```diff
@@ -0,0 +1,35 @@
...
+            ]
+        }]
+    }
+    ```
+
+# License
+The Edge Orchestration source code is distributed under the Apache License, Version 2.0 open source license.
+
+# Contributing
+If you want to contribute to the Edge Orchestration project and make it better, your help is very welcome. Contributing is also a great way to learn more about social coding on Github, new technologies and and their ecosystems.
```
Comment (MoonkiHong) [inline README.md]:
```suggestion
If you want to contribute to the Edge Orchestration project and make it better, your help is very welcome. Contributing is also a great way to learn more about social coding on GitHub, new technologies and their ecosystems.
```

**Review by MoonkiHong — COMMENTED** [review]
PTAL. 😄

**Review by tiokim — CHANGES_REQUESTED** [review]
https://github.com/lf-edge/edge-home-orchestration-android/pull/1/commits/dfb6db74ac503c5c7257dff30ea8004f5828410c makes weird.

**Comment by suresh-lc** [comment]
> [dfb6db7](https://github.com/lf-edge/edge-home-orchestration-android/commit/dfb6db74ac503c5c7257dff30ea8004f5828410c) makes weird.

Hope fine now

**Review by tiokim — APPROVED** [review]

**Review by MoonkiHong — APPROVED** [review]
+1

**Review by tdrozdovsky — APPROVED** [review]
Great first step!

**Pull request merged**